pkg/cache/snapshot.go

```diff
@@ -173,7 +187,27 @@ func (c *ClusterQueue) accumulateResources(cohort *Cohort) {
 			cohort.Usage[fName] = used
 		}
 		for res, val := range resUsages {
+			//val -= c.ReservedQuota(fName, res)
+			//if val < 0 {
+			//	val = 0
+			//}
 			used[res] += val
 		}
 	}
 }
+
+// CohortUsage recalculates cohort.Usage when LendingLimit enabled,
+// it represents the quota cq is using over its GuaranteedQuota.
+func (c *ClusterQueue) CohortUsage(fName kueue.ResourceFlavorReference, rName corev1.ResourceName) (val int64) {
+	cohortUsage := c.Cohort.Usage[fName][rName]
+
+	// When feature LendingLimit enabled, Cohort.Usage contains the sum of
+	// all CQs' GuaranteedQuota, so we need to remove the guaranteed resources.
+	cohortUsage -= c.ReservedQuota(fName, rName)
```

Comment:
this comment doesn't look accurate at all.

I don't understand the rationale of the code overall. And it looks under-tested. Please fix comments and add more test scenarios, specially in shceuler_test.go.

Reply:
Done.